ntoskrnl/include/internal/mm.h

```diff
@@ -866,6 +866,36 @@ NTAPI
 MmDeleteKernelStack(PVOID Stack,
                     BOOLEAN GuiStack);
 
+/* balance.c / pagefile.c******************************************************/
+
+inline VOID UpdateTotalCommittedPages(LONG Increase)
+{
+    /*
+     * Add up all the used "Committed" memory + pagefile.
+     * Not sure this is right. 8^\
+     * MmTotalCommittedPages should be adjusted consistently with
+     * other counters at different places.
+         MmTotalCommittedPages = MiMemoryConsumers[MC_SYSTEM].PagesUsed +
+                            MiMemoryConsumers[MC_USER].PagesUsed +
+                            MiUsedSwapPages;
+     
+     */
+    
+    /* Update Commitment */
+    SIZE_T TotalCommittedPages = InterlockedExchangeAddSizeT(&MmTotalCommittedPages, Increase) + Increase;
+
+    SIZE_T PeakCommitment = MmPeakCommitment;
+    
+    if (Increase < 0) return; //if we decrement, we exit here to optimize processing (peak not impacted)
```

Comment:
If that's a nagative, yout guaranteed not to enter the while loop below.

Reply:
Done